pkg/console/subresource/configmap/configmap.go

```diff
@@ -66,25 +66,34 @@ func getApiUrl(infrastructureConfig *configv1.Infrastructure) string {
 // - a bool indicating if config was merged (unsupportedConfigOverrides)
 // - an error
 func DefaultConfigMap(operatorConfig *operatorv1.Console, consoleConfig *configv1.Console, managedConfig *corev1.ConfigMap, infrastructureConfig *configv1.Infrastructure, rt *routev1.Route) (*corev1.ConfigMap, bool, error) {
+	// There are multiple sources for the config
+	//1. default brand `okd`, as well as default doc url & logout url set
+	//2. extract managed config and apply on top of this, without nilling out prev values
+	//3. apply operator config on top of this, without nilling out prev values
+	//4. apply unsupported config overrides on top of this, but should not nil out prev values
+
+	// Build a default config that can be overwritten from other sources
+	host := rt.Spec.Host
+	apiServerURL := getApiUrl(infrastructureConfig)
+	defaultConfig := NewYamlConfig(host, defaultLogoutURL, DEFAULT_BRAND, DEFAULT_DOC_URL, apiServerURL)
+
+	// Derive a config using operator and console configs
 	logoutRedirect := getLogoutRedirect(consoleConfig)
 	brand := getBrand(operatorConfig)
 	docURL := getDocURL(operatorConfig)
-	apiServerURL := getApiUrl(infrastructureConfig)
-
-	host := rt.Spec.Host
-	config := NewYamlConfig(host, logoutRedirect, brand, docURL, apiServerURL)
+	derivedConfig := NewYamlConfig(host, logoutRedirect, brand, docURL, apiServerURL)
 
-	configMap := Stub()
-	configMap.Data = map[string]string{}
 	unsupportedRaw := operatorConfig.Spec.UnsupportedConfigOverrides.Raw
-	newConfig := extractYAML(managedConfig)
+	// Get console config from the openshift-config-managed namespace
+	managedConfigYAML := extractYAML(managedConfig)
 
-	// merge config overrides, if we have them
-	mergedConfig, err := resourcemerge.MergeProcessConfig(nil, config, newConfig, unsupportedRaw)
+	// merge configs with overrides, if we have them
```

Comment:
So naming, I may prefer dropping `derived` in favor of pretty specific var names:

```go
defaultConfig :=NewYamlConfig(host, defaultLogoutURL, DEFAULT_BRAND, DEFAULT_DOC_URL, apiServerURL)
extractedManagedConfig := extractYAML(managedConfig)
operatorConfig := NewYamlConfig(host, logoutRedirect, brand, docURL, apiServerURL)
unsupportedConfigOverride := operatorConfig.Spec.UnsupportedConfigOverrides.Raw

mergedConfig, err := resourcemerge.MergeProcessConfig(nil, defaultConfig, extractedManagedConfig, operatorConfig, unsupportedConfigOverride)
```

Reply:
I agree with all, except the name `operatorConfig` is already passed in so I will call it `consoleOperatorConfig`.

Reply:
I would disambiguate further then.  `operator` config and `console operator` config still imply the same thing (we have only 1 operator).

Reply:
You're trying to say "generated console config derived from the operator config" but still a temp version until after the `merge`.

Reply:
You could go with something like:

```

defaultConfig
extractedManagedConfig
userDefinedConfig  // because this is derived from user intent via explicit user config
unsupportedRawOverrides
```

Reply:
userDefined - I think that hits the mark, I will make the change